chapters/3/modul-kompilujacy.tex

```diff
@@ -0,0 +1,75 @@
+\section{Moduł kompilujący}
+W czasie prac nad projektem nie posiadaliśmy dostępu do istniejącej części systemu modułu kompilującego, odpowiedzialnego za kompilację kodu, egzekucję programu oraz przeprowadzanie testów. Znane były funkcjonalności oraz wymagania, które musiał spełniać ten moduł, co pozwoliło nam na implementację własnego modułu. Rozwiązanie w obecnym stanie było używane jedynie do testów, jednak jest otwarte na modyfikację i dalszy rozwój w przyszłości.
+
+\subsection{Wymagania}
+Otrzymane wymagania odnośnie modułu kompilującego obejmowały wykorzystane technologie oraz sposób działania. Moduł musiał być skonteneryzowany, zapewniając izolację wykonywanego kodu zadania od systemu operacyjnego i procesów hosta. Ze względu na dostępność wiedzy, łatwość implementacji oraz utrzymywania systemu, oprogramowaniem służącym do konteneryzacji miał być Docker. Został również określony wymóg, aby czas wykonania zadania był niezależny od obciążenia i dostępnej mocy obliczeniowej systemu hosta. Dodatkowo system miał zapewniać wykonywanie programu zadania tak, jakby był wykonywany na maszynie z systemem operacyjnym Windows. Ważnym aspektem był również niski koszt związany z potencjalną awarią i ponownym przywróceniem działania tej części systemu.
+
+\subsection{Obraz bazowy}
+Wybranym obrazem bazowym do kontenera zostało Ubuntu 20.04 \cite{linuxUbuntu}, jest to popularny wybór, dzięki czemu jest dobrze przetestowany i kompatybilny z wieloma narzędziami. Obrazy Ubuntu dostępne w Docker Hub to systemy oparte na Ubuntu Base, czyli lżejszej wersji Ubuntu, nieposiadającej oprogramowania, które nie jest przydatne do pracy w kontenerze takich jak interfejs graficzny lub aplikacje biurowe. Wersja 20.04 jest wersją Long-Term Support, co oznacza, że przez długi czas oferuje aktualizacje bezpieczeństwa oraz poprawki. Posiada również bogate repozytorium pakietów, oferujące wszystkie potrzebne w naszym zastosowaniu narzędzia.
```

Comment:
> Wybranym obrazem bazowym do kontenera zostało Ubuntu 20.04 \cite{linuxUbuntu}, jest to popularny wybór, dzięki czemu jest dobrze przetestowany i kompatybilny z wieloma narzędziami.

Ale dlaczego my go wybraliśmy? Można się domyśleć, że popularność, ale imo lepiej powiedzieć wprost

> które nie jest przydatne do pracy w kontenerze takich jak interfejs graficzny lub aplikacje biurowe.

To jest bardzo subiektywne stwierdzenie. Trochę bym się rozpisał bardziej o tych różnicach.

> Wersja 20.04 jest wersją Long-Term Support, co oznacza, że przez długi czas oferuje aktualizacje bezpieczeństwa oraz poprawki.

Nie wiemy do kiedy? wydawało mi się że canonical (tfu) wypisuje do kiedy jest support

> Posiada również bogate repozytorium pakietów, oferujące wszystkie potrzebne w naszym zastosowaniu narzędzia.

Ciekawostka jest taka, że apt jest stosunkowo biedny jak na repo pakietów. Warto też wspomnieć że korzystamy właśnie z apt, bo samo ubuntu ma zainstalowane 3 różne menadzery

Reply:
> To jest bardzo subiektywne stwierdzenie. Trochę bym się rozpisał bardziej o tych różnicach.

Tu imo wystarczy jedno zdanie